plugin/plugins/postgresds/postgresds.go

```diff
@@ -0,0 +1,110 @@
+package postgresds
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	postgresdb "gx/whyrusleeping/QmXyfyKbv3EkZUtN5oxL2c32HNPaJKYsCQhxPufMPxKwzi/sql-datastore/postgres"
+
+	"github.com/ipfs/go-ipfs/plugin"
+	"github.com/ipfs/go-ipfs/repo"
+	"github.com/ipfs/go-ipfs/repo/fsrepo"
+)
+
+// Plugins is exported list of plugins that will be loaded
+var Plugins = []plugin.Plugin{
+	&postgresdsPlugin{},
+}
+
+type postgresdsPlugin struct{}
+
+var _ plugin.PluginDatastore = (*postgresdsPlugin)(nil)
+
+func (*postgresdsPlugin) Name() string {
+	return "ds-postgresds"
+}
+
+func (*postgresdsPlugin) Version() string {
+	return "0.1.0"
+}
+
+func (*postgresdsPlugin) Init() error {
+	return nil
+}
+
+func (*postgresdsPlugin) DatastoreTypeName() string {
+	return "postgres"
+}
+
+type datastoreConfig struct {
+	host     string
+	user     string
+	passfile string
+	password string
+	dbname   string
+}
+
+// BadgerdsDatastoreConfig returns a configuration stub for a badger datastore
+// from the given parameters
+func (*postgresdsPlugin) DatastoreConfigParser() fsrepo.ConfigFromMap {
+	return func(params map[string]interface{}) (fsrepo.DatastoreConfig, error) {
+		var c datastoreConfig
+		var ok bool
+		c.host, ok = params["host"].(string)
+		if !ok {
+			return nil, fmt.Errorf("'path' field was not a string")
+		}
+		c.user, ok = params["user"].(string)
+		if !ok {
+			return nil, fmt.Errorf("'user' field was not a string")
+		}
+		c.passfile, ok = params["passfile"].(string)
+		if !ok {
+			return nil, fmt.Errorf("'passfile' field was not a string")
+		}
+		c.dbname, ok = params["dbname"].(string)
+		if !ok {
+			return nil, fmt.Errorf("'dbname' field was not a string")
+		}
+		c.password, ok = params["password"].(string)
+		if !ok {
+			return nil, fmt.Errorf("'password' field was not a string")
+		}
+		return &c, nil
+	}
+}
+
+func (c *datastoreConfig) DiskSpec() fsrepo.DiskSpec {
+	return map[string]interface{}{
+		"type":     "postgres",
+		"user":     c.user,
+		"database": c.dbname,
+	}
+}
+
+func (c *datastoreConfig) Create(path string) (repo.Datastore, error) {
+	if c.passfile != "" {
```

Comment:
This should happen in `DatastoreConfigParser` so that `DiskSpec` writes the right thing